serial_server.py

```diff
@@ -53,28 +56,62 @@ class NoPortsAvailableError(Error):
     """No serial ports are available."""
     code = 3
 
+SerialDevice = collections.namedtuple('SerialDevice', ['name', 'devicepath'])
 
 class SerialServer(LabradServer):
     """Provides access to a computer's serial (COM) ports."""
     name = '%LABRADNODE% Serial Server'
 
     def initServer(self):
+        if sys.platform.startswith('win32'):
+            self.enumerate_serial_windows()
+        else:
+            self.enumerate_serial_pyserial()
+            
+    def enumerate_serial_windows(self):
+        """This manually attempts to enumerate the first 20 COM ports.
+        pyserial includes a function to enumerate device names, but it
+        possibly doesn't work right on windows for COM ports above 4.
+        http://stackoverflow.com/questions/12090503/listing-available-com-ports-with-python
+        """
         self.SerialPorts = []
         print 'Searching for COM ports:'
         for a in range(1, 20):
             COMexists = True
             try:
-                ser = Serial('\\\\.\\COM%d' % a)
+                ser = Serial(r'\\.\COM%d' % a)
                 ser.close()
-            except SerialException, e:
+            except SerialException as e:
                 if e.message.find('cannot find') >= 0:
                     COMexists = False
             if COMexists:
-                self.SerialPorts += ['COM%d' % a]
+                self.SerialPorts.append(SerialDevice('COM%d' % a, r'\\.\COM%d' % a))
                 print '  COM%d' % a
         if not len(self.SerialPorts):
             print '  none'
 
+    def enumerate_serial_pyserial(self):
+        """This uses the pyserial built-in device enumeration.  However,
+        we ignore the pyserial "human readable" device name because that
+        appears to make no sense.  For instance, a particular FTDI USB-Serial
+        adapter shows up as 'Microsoft Corp. Optical Mouse 200'.
+
+        Following the example from the above windows version, we try to open
+        each port and ignore it if we can't.
+        """
+        dev_list = serial.tools.list_ports.comports()
+        self.SerialPorts = []
```

Comment:
would be nice to rename this as `serial_ports`, but that may be out of scope for this change.